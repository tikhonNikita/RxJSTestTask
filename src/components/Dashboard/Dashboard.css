.dashboard {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  height: 100vh;
  background: linear-gradient(38deg, #bbc9d5, #8dd490);
  font-family: 'Roboto', sans-serif;
}

.temperature,
.pressure,
.humidity {
  box-shadow: 0px 0px 44px 0px rgba(0, 0, 0, 0.75);
  width: 50%;
  margin: 2%;
  border-radius: 10px;
  background: white;
}
