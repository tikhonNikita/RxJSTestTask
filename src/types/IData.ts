export default interface IData {
    temperature: string;
    pressure: string;
    humidity: string
}