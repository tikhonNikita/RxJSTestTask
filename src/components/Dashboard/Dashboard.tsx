import React from 'react';

import IData from '../../types/IData';
import './Dashboard.css';

export const Dashboard: React.FC<IData> = ({
  temperature,
  pressure,
  humidity
}) => {
  return (
    <div className="dashboard">
      <div className="temperature">
        <h1 className="dashboard__h1">Temperature</h1>
        <div className="temperature__h2">
          <h2>{temperature}</h2>
        </div>
      </div>
      <div className="pressure">
        <h1 className="dashboard__h1">Pressure</h1>
        <div className="pressure__h2">
          <h2>{pressure}</h2>
        </div>
      </div>
      <div className="humidity">
        <h1 className="dashboard__h1">Humidity</h1>
        <div className="humidity__h2">
          <h2>{humidity}</h2>
        </div>
      </div>
    </div>
  );
};
