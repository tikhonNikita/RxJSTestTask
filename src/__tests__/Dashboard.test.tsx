import React from 'react';
import { create } from 'react-test-renderer';

import IData from '../types/IData';
import { Dashboard } from '../components/Dashboard/Dashboard';

describe('renders dashboard component', () => {
  test('Matches the snapshot', () => {
    const testData: IData = {
      temperature: '123',
      pressure: '3',
      humidity: '10'
    };
    const dashboard = create(<Dashboard {...testData} />);

    expect(dashboard.toJSON()).toMatchSnapshot();
  });
});
