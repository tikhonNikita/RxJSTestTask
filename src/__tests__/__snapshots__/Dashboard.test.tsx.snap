// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dashboard component Matches the snapshot 1`] = `
<div
  className="dashboard"
>
  <div
    className="temperature"
  >
    <h1
      className="dashboard__h1"
    >
      Temperature
    </h1>
    <div
      className="temperature__h2"
    >
      <h2>
        123
      </h2>
    </div>
  </div>
  <div
    className="pressure"
  >
    <h1
      className="dashboard__h1"
    >
      Pressure
    </h1>
    <div
      className="pressure__h2"
    >
      <h2>
        3
      </h2>
    </div>
  </div>
  <div
    className="humidity"
  >
    <h1
      className="dashboard__h1"
    >
      Humidity
    </h1>
    <div
      className="humidity__h2"
    >
      <h2>
        10
      </h2>
    </div>
  </div>
</div>
`;
